lib/web/apiserver.go

```diff
@@ -521,23 +521,23 @@ func (h *Handler) bindDefaultEndpoints() {
 	// endpoint returns the default authentication method and configuration that
 	// the server supports. the /webapi/ping/:connector endpoint can be used to
 	// query the authentication configuration for a specific connector.
-	h.GET("/webapi/ping", httplib.MakeHandler(h.ping))
-	h.GET("/webapi/ping/:connector", httplib.MakeHandler(h.pingWithConnector))
+	h.GET("/webapi/ping", h.WithLimiter(h.ping))
```

Comment:
I would be a little bit concerned about limiting the ping endpoint, I think it may be used as part of the node joining or initial connection process. (we should confirm whether this is the case or not)

For example, suppose a customer has thousands of nodes in some private network with SNAT. It will appear as if all of these nodes are connecting from the same source IP and could trip the rate limiter and prevent nodes from joining or reconnecting after an upgrade.

Reply:
https://github.com/gravitational/teleport/issues/4330 specifically calls out the file descriptors that are opened from ping.  I understand the concern though.

In such a condition, would the reconnecting nodes retry until they are able to eventually get a rate allowed request?

Do you think any rate limit could be designed around ping?

Reply:
@zmb3 As an alternative solution, could the nodes start to send their `ping` request authenticated?  If so we could still make this endpoint "optionally unauthenticated", but let the rate limiter ignore requests which provide valid authentication.

Reply:
I added the ability to only rate limit unauthenticated requests in this commit: https://github.com/gravitational/teleport/pull/24623/commits/289b6114f5e6acd4749be8ef0bd046881ec077dd

As mentioned in the message luckily `ping` was already providing authentication, so this logic worked pretty straight forward out of the box.  I added this to other endpoints which _might_ be good candidates to consider, although I did not verify they were providing auth like `ping` was.

Please provide feedback on this change, thank you!